src/main/java/CancelApplication/CancelApplicationInteractor.java

```diff
@@ -0,0 +1,61 @@
+package CancelApplication;
+
+import Entities.Group;
+import Entities.User;
+
+/**
+ * The cancelApplication use case.
+ * The user can only cancel applications for groups that they have previously applied to,
+ * where the user has not already been rejected or approved.
+ */
+
+public class CancelApplicationInteractor implements CancelApplicationInputBoundary {
+
+    final CancelApplicationDsGateway dsGateway;
+    final CancelApplicationOutputBoundary outputBoundary;
+
+    /**
+     * The interactor for the cancelApplication use case.
+     * @param dsGateway the data access interface
+     * @param outputBoundary the output boundary implemented by the cancelApplicationPresenter
+     */
+    public CancelApplicationInteractor(CancelApplicationDsGateway dsGateway,
+                                       CancelApplicationOutputBoundary outputBoundary) {
+        this.dsGateway = dsGateway;
+        this.outputBoundary = outputBoundary;
+    }
+
+    /**
+     * @param requestModel the requestModel for the cancelApplication use case
+     * @return a successView or failView depending on whether the user has cancelled their group application
+     */
+    @Override
+    public CancelApplicationResponseModel cancelApplication(CancelApplicationRequestModel requestModel) {
+
+        if (!dsGateway.groupExists(requestModel.getGroupname())) {
+            return outputBoundary.prepareFailureView("Group does not exist.");
+        }
+
+        User user = dsGateway.getUser(requestModel.getUsername());
+        Group group = dsGateway.getGroup(requestModel.getGroupname());
```

Comment:
One thing that I was wondering is that should we get the user this way from the gateway. I had initialized the user through CurrentUser.getInstance().getUser(), does this approach make sense or should I also change mine?